trackpy/motion.py

```diff
@@ -184,7 +184,10 @@ def subtract_drift(traj, drift=None):
 
     if drift is None:
         drift = compute_drift(traj)
-    return traj.set_index('frame', drop=False).sub(drift, fill_value=0)
+    traj.set_index('frame', inplace=True, drop=False)
+    for col in drift.columns:
```

Comment:
Ah, that's right, we don't need `.assign` for this. If you don't mind benchmarking I'll wait to merge. If you don't want to bother I'm fine with merging -- you are right that's a small loop.